### PR TITLE
refactor: use ValueTask in cached scenarios

### PR DESCRIPTION
**Pull request by linkdotnet**
The Pull Request adds optimization in scenarios where the cache is hit, and the value doesn't have to be evaluated, leading to a smaller footprint and an overall more expressive API.

### CODE REVIEW

**Comment by timonkrebs** [comment]
Thank you for the contribution.

I do not want to expose Value Task in any way from this lib. As Signals should be able to be read multiple times.

But at the moment I do not see why Stale would be exposed in the Future. Therefor LGTM.

I still would like to think about introducing Value Task a bit more.

**Comment by linkdotnet** [comment]
It’s absolutely fine to keep the PR around without merging and tacking it later one

File: MemoizR/EagerRelativeSignal.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public async Task Set(Func<T?, T> fn)
         }
     }
 
-    public async Task<T?> Get()
+    public async ValueTask<T?> Get()
     {
```
Comment (timonkrebs) [inline MemoizR/EagerRelativeSignal.cs:36]: I do not want to expose ValueTask here as it is expected that Get() will be evaluated multiple times.

Comment (timonkrebs) [reply MemoizR/EagerRelativeSignal.cs:36]: > Valid consumption patterns for ValueTasks
> Awaiting a ValueTask / ValueTask<TResult> multiple times. The underlying object may have been recycled already and be in use by another operation. In contrast, a Task / Task<TResult> will never transition from a complete to incomplete state, so you can await it as many times as you need to, and will always get the same answer every time.
https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/

Comment (linkdotnet) [reply MemoizR/EagerRelativeSignal.cs:36]: Fair - One might pass down a Signal down the component tree in a UI Framework.

**Pull request closed without merging**